tcms/rpc/tests/test_testcase.py

```diff
@@ -610,3 +700,49 @@ def verify_api_with_permission(self):
     def verify_api_without_permission(self):
         with self.assertRaisesRegex(ProtocolError, '403 Forbidden'):
             self.rpc_client.TestCase.comments(self.case.pk)
+
+
+class TestAddAttachment(APIPermissionsTestCase):
+    permission_label = 'attachments.add_attachment'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.case = TestCaseFactory()
+
+    def verify_api_with_permission(self):
+        self.rpc_client.TestCase.add_attachment(self.case.pk,
+                                                'test.txt',
+                                                'test')
+        attachments = Attachment.objects.attachments_for_object(self.case)
+        self.assertEqual(len(attachments), 1)
+        self.assertEqual(attachments[0].object_id, str(self.case.pk))
+
+    def verify_api_without_permission(self):
+        with self.assertRaisesRegex(ProtocolError, '403 Forbidden'):
+            self.rpc_client.TestCase.add_attachment(self.case.pk,
+                                                    'test.txt',
+                                                    'test')
+
+class TestListAttachment(APIPermissionsTestCase):
+    permission_label = 'attachments.view_attachment'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.case = TestCaseFactory()
+        add_attachment(self.case.pk,
+                       'testcases.TestCase',
+                       self.tester,
+                       'test.txt',
+                       'test')
```

Comment:
@atodorov how should I add an attachment here. The attachments does not shows up when the code below. What am I missing here? 

I looked into the implementation of `Testcase.add_attachment` and it used this method  `tcms.rpc.utils.add_attachment`.  I am calling this method as it would have been called in `Testcase.add_attachment`. 
Using an API method here does not makes any sense as you shared in your previous review.

Reply:
Other than the fact that the last argument should be base64 encoded I don't see anything else that may be wrong. Maybe there is a silent failure here.  See how data is passed in  tests/test_testplan.py from the API method to the utils function.